core/src/main/java/com/linecorp/armeria/common/HttpData.java

```diff
@@ -386,6 +387,10 @@ default String toStringAscii() {
 
     /**
      * Returns a new {@link InputStream} that is sourced from this data.
+     *
+     * <p>Note, if this {@link HttpData} is pooled (e.g., it is the result of a call to
+     * {@link HttpResponse#aggregateWithPooledObjects(ByteBufAllocator)}), then this {@link InputStream} will
+     * release the underlying buffer back to the pool when closed.
```

Comment:
is there impact to calling release twice? or this is mainly to let people know that they are effectively transferring responsibility of the http data to this inputstream and shouldn't use it later

Reply:
Netty will throw an exception if calling release too many times, meaning the reference count goes negative. The comment is to let users know that they don't need to release, though if they need to keep the buffer anyways they need to first `retain()`. Using pooled objects is definitely more advanced :)

Reply:
thx will keep in mind for when I use this.

Reply:
Shouldn't we retain the buffer before creating a `ByteBufInputStream`? I find it unnatural that `HttpData` is released when its stream is closed, because otherwise converting an `HttpData` into an `InputStream` twice might not work:

```java
sendTo(data.toInputStream(), outStreamA);
sendTo(data.toInputStream(), outStreamB);
data.release();
```